format/postgres/flavours/postgres14/common14/pgheap.go

```diff
@@ -0,0 +1,492 @@
+			d.SeekAbs(itemPos)
+			itemId.lpLen = uint32(d.FieldU32("lp_len", common.LpLenMapper))
+
+			page.ItemIds = append(page.ItemIds, itemId)
+		})
+	} // for pd_linp
+}
+
+func decodeTuples(d *decode.D) {
+	heap := GetHeapD(d)
+	page := heap.Page
+
+	for i := 0; i < len(page.ItemIds); i++ {
+		id := page.ItemIds[i]
+		if id.lpOff == 0 || id.lpLen == 0 {
+			continue
+		}
+		if id.lpFlags != common.LP_NORMAL {
+			continue
+		}
+
+		pos := int64(page.PagePosBegin)*8 + int64(id.lpOff)*8
+		tupleDataLen := id.lpLen - SizeOfHeapTupleHeaderData
+
+		// seek to tuple with ItemId offset
+		d.SeekAbs(pos)
+
+		// type = struct HeapTupleHeaderData {
+		/*    0      |    12 */ // union {
+		/*                12 */ //     HeapTupleFields t_heap;
+		/*                12 */ //     DatumTupleFields t_datum;
+		//						} t_choice;
+		/* total size (bytes):   12  */
+		//
+		/*   12      |     6 */ // ItemPointerData t_ctid;
+		/*   18      |     2 */ // uint16 t_infomask2;
+		/*   20      |     2 */ // uint16 t_infomask;
+		/*   22      |     1 */ // uint8 t_hoff;
+		/*   23      |     0 */ // bits8 t_bits[];
+		/* XXX  1-byte padding  */
+		//
+		/* total size (bytes):   24 */
+		d.FieldStruct("HeapTupleHeaderData", func(d *decode.D) {
+
+			pos1 := d.Pos()
+			// we need infomask before t_xmin, t_xmax
+			d.SeekAbs(pos1 + 18*8)
+			d.FieldU16("t_infomask2")
+			d.FieldStruct("Infomask2", decodeInfomask2)
+			d.FieldU16("t_infomask")
+			d.FieldStruct("Infomask", decodeInfomask)
+
+			// restore pos and continue
+			d.SeekAbs(pos1)
+			d.FieldStruct("t_choice", decodeTChoice)
+			d.FieldStruct("t_ctid", func(d *decode.D) {
+				/*    0      |     4 */ // BlockIdData ip_blkid;
+				/*    4      |     2 */ // OffsetNumber ip_posid;
+				d.FieldU32("ip_blkid")
+				d.FieldU16("ip_posid")
+			}) // ItemPointerData t_ctid
+
+			/*   18      |     2 */ // uint16 t_infomask2;
+			/*   20      |     2 */ // uint16 t_infomask;
+			/*   22      |     1 */ // uint8 t_hoff;
+			/*   23      |     0 */ // bits8 t_bits[];
+			/* XXX  1-byte padding  */
+			//d.FieldU16("t_infomask2")
+			//d.FieldStruct("Infomask2", decodeInfomask2)
+			//d.FieldU16("t_infomask")
+			//d.FieldStruct("Infomask", decodeInfomask)
+			// already done
+			d.SeekRel(32)
+
+			d.FieldU8("t_hoff")
+			d.U8()
+
+			d.FieldRawLen("t_bits", int64(tupleDataLen*8), scalar.RawHex)
+
+		}) // HeapTupleHeaderData
+	} // for ItemsIds
+}
+
+func decodeInfomask2(d *decode.D) {
+	heap := GetHeapD(d)
+	tuple := &TupleD{}
+	heap.Tuple = tuple
+
+	pos := d.Pos() - 16
+	d.SeekAbs(pos)
+	d.FieldU16("HEAP_KEYS_UPDATED", common.Mask{Mask: HEAP_KEYS_UPDATED})
+	d.SeekAbs(pos)
+	d.FieldU16("HEAP_HOT_UPDATED", common.Mask{Mask: HEAP_HOT_UPDATED})
+	d.SeekAbs(pos)
+	d.FieldU16("HEAP_ONLY_TUPLE", common.Mask{Mask: HEAP_ONLY_TUPLE})
+}
+
+func decodeInfomask(d *decode.D) {
+	heap := GetHeapD(d)
+	tuple := heap.Tuple
+
+	pos := d.Pos() - 16
+	d.SeekAbs(pos)
+	d.FieldU16("HEAP_HASNULL", common.Mask{Mask: HEAP_HASNULL})
```

Comment:
This seems to be individual bit flags? could this be:
```go
d.FieldBool("HEAP_HASVARWIDTH",)
d.FieldBool("HEAP_HASEXTERNAL")
d.FieldBool("HEAP_HASOID_OLD")
...
```
And possibly add "unused" or "reserved" fields for bit that are not used